### ISSUE TITLE
[netrunner]: choosing another translation language in system settings does not trigger downloading langauge packs

### ISSUE THREAD

**Issue by star-buck** [blue-systems/plasma-5.3#74]
ISO: http://dev.netrunner-os.com/dev/netrunner-15-dvd-64bit-build7g.iso

**Comment by jriddell** [comment]
The langpack install patch has not been ported to Plasma 5 so it won't work currently.  The translations kcm has been rewritten for Plasma 5 (by John Layt) who did add buttons and empty methods for language install but left it up to distros to fill that in which is still on the todo list for kubuntu.

**Comment by jriddell** [comment]
here's a patch for plasma-desktop which replaces the kcm translations with one that just loads kcmshell4 locale http://starsky.19inch.net/~jr/tmp/kcm-translations-load-kde4.diff

**Comment by star-buck** [comment]
@jriddell / @apachelogger : Do we still need this patch for vivid-based netrunner/kci?

**Comment by star-buck** [comment]
Currently, after picking German in ubiquity installer, some of the translations are missing and Translations KCM looks like this:
![language-qt5-nontranslated](https://cloud.githubusercontent.com/assets/840070/7554128/cea8afda-f71b-11e4-9af2-f313072ad5bd.png)

**Comment by apachelogger** [comment]
@star-buck yes there still is no plasma5 port of the patch. Translations are probably missing because CI builds have no translations.

**Comment by star-buck** [comment]
@llelectronics :
Thing is kci ppa stable had 5.3+git as package versions, so simply going back from netrunner-16-ci-packages to a 5.3 "lower number" in current ISO isnt easily possible without risking to test it all over again...?
@apachelogger : Can we nevertheless ship translations packages? They must be somehwere and is it possible to have them recognized?
Otherwise, I'd suggest copy P5.3, Frameworks5.10 and apps 15.04 official releases here for testing it out:
https://launchpad.net/~netrunner-os/+archive/ubuntu/netrunner-16-official

**Comment by llelectronics** [comment]
For testing in current ISO it is a little bit cumbersome as you manually have to downgrade the packages then and of course test if it does not bring new bugs

**Comment by apachelogger** [comment]
@star-buck there are no "translation packages". translations are part of the upstream plasma tarballs and consequently also part of the application debs. So, just like with netrunner 15, we would have needed to use the release packages Plasma 5.3 and work from there rather than use the CI packages. I suppose my question about exactly this problem got lost in the IRC backlog last week :|.

There's a couple of ways to go about this, all of them are extremely crappy first one should be the least awful though.

- Downgrade Plasma packages to 5.3 as suggested. I have no clue how the ISO is created but forcing the ISO to the lower version should in the worst case only be a case of setting an apt preference pin score >1000 on a PPA that only holds 5.3 which would then make apt downgrade all packages from that PPA regardless of their version. Then we just need to adjust the fixes PPA to use the correct versions. That's pretty much 3 lines in some apt config and then we just need to repackage the plasma bits in the fixes PPA. The general process is outlined here https://apachelog.wordpress.com/2014/01/16/neon-5s-many-ppas-apt/ . Since our current packages are from exactly the tagging date this has next to no regression potential.
- Repackage every single plasma CI tarball and add the translations from the plasma 5.3 release tarball. This might well be approach with the least regression potential because the source code remains exactly the same. There is some additional logic involved in building the release tarballs though which might mean that we will have to adjust the cmake code to get building packages.
- Take plasma 5.3 tarballs and extract the translations, pack them together into a new tarball and use that to package translations. This would likely require substantial adjustment to the cmake parts involved and essentially make this one translation package conflict with every single Kubuntu plasma package as they'd have overlapping files; making this a very scary approach that would eventually probably fall us on the head.

**Comment by star-buck** [comment]
so for apt pinning an official releases ppa>1000:
after downgrading packages on the ISO, cant we simply remove the "ci" ppa then and after that also remove the high pinning of the official ppa, so fixed packages from ci-fixes would get applied normally?

**Comment by star-buck** [comment]
@apachelogger : after downgrading to official Plasma 5.3:
Any idea when this patch gets ported to work?

**Comment by apachelogger** [comment]
no clue
